src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -70,11 +71,27 @@ internal static bool IsDirectoryTooLong(string fullPath)
         }
 
         /// <summary>
-        /// Adds the extended path prefix (\\?\) if not already present.
+        /// Adds the extended path prefix (\\?\) if not already present, IF the path is not relative,
+        /// AND the path is more than 259 characters. (> MAX_PATH + null)
+        /// </summary>
+        internal static string AddExtendedPathPrefixForLongPaths(string path)
```

Comment:
(Reading it again, I think that might be what you were going for with "ForLongPaths".  FWIW, I didn't interpret it to mean "will add the prefix only if it's necessary to support a long path" but rather "always add the prefix that's used with long paths".)